test/models/user_test.rb

```diff
@@ -0,0 +1,8 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+  test "should not save article without token or username" do
+  user = User.new
+  assert_not user.save
```

Comment:
I don't think this is really checking what the test description says. It's only checking that `user.save` returns `nil` or `false`. Although I don't think these tests are really necessary, since it's basically testing that Rails validations work like do, if you really want to add those tests, you can do something like:
```ruby
test "should not save a gen without a username or a token or an email" do
  user = User.new
  user.save
  assert_includes user.errors, :username
  assert_includes user.errors, :token
  assert_includes user.errors, :email
end
```